src/operation/aggregate/mod.rs

```diff
@@ -91,6 +92,12 @@ impl Operation for Aggregate {
             .and_then(|opts| opts.selection_criteria.as_ref())
     }
 
+    fn supports_read_concern(&self, description: &StreamDescription) -> bool {
```

Comment:
Does this have to do with causal consistency or is this a validation we were missing?

Reply:
Causal consistency requires that we set a read concern in the executor now, whereas before individual operations were in charge of serializing read concerns. Now that the executor is in charge of that, it needs to know when it can set them, hence this method. The check for it can be seen here: https://github.com/awitten1/mongo-rust-driver/blob/RUST-48_Causal-Consistency/src/client/executor.rs#L430

I think users setting read concerns on operations that don't support them is one of those things we're meant to leave to the server to validate, but we do need to validate that the driver doesn't accidentally set a read concern behind the scenes when it shouldn't.